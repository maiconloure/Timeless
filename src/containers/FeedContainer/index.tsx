/* eslint-disable @typescript-eslint/no-unused-vars */
import React from 'react';
import { useSelector } from 'react-redux';

import { NotificationFeed } from '../../components';
import { RootStoreType } from '../../redux/store/store';

const FeedContainer = () => {
  const currentBoard = useSelector((state: RootStoreType) => state.boards.currentBoard);
  return (
    <NotificationFeed
      prop={currentBoard && currentBoard.data ? currentBoard.data.notifications : []}
    />
  );
};

export default FeedContainer;
