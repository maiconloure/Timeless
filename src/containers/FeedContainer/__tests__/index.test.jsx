import React from 'react';
import { useSelector } from 'react-redux';
import renderer from 'react-test-renderer';
import FeedContainer from '../index';

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}));

describe('Renders', () => {
  beforeEach(() => {
    useSelector.mockImplementation(() => ['test1', 'test2']);
  });

  it('add new messages', () => {
    const tree = renderer.create(<FeedContainer prop={useSelector} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
