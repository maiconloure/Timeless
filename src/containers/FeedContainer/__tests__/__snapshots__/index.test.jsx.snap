// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders add new messages 1`] = `
<div
  className="sc-hiKfjK kmjzEC"
  height="450px"
  width="280px"
>
  <div
    className="sc-gXfWUo cTZZDr"
  >
    <div
      className="sc-cBoprd hXbwCO"
    >
      <h2>
        Notificações recentes
      </h2>
    </div>
    <div
      className="sc-ciSmjq jLvfax"
      onClick={[Function]}
    >
       - 
    </div>
  </div>
  <div
    className="sc-jcwofb sc-carGAA kNhOZX gXYsNS"
  />
</div>
`;
